### PR TITLE
Add using OutputFormat enum to --sign-only transactions

### LINKED ISSUE #8281: Redundant output from --sign-only
#### Problem

Commands with the `--sign-only` option output both human-friendly text and JSON.

#### Proposed Solution

It should output one or the other, depending on the presence of a `--json` (or similar name) option.

### PR DESCRIPTION
**Pull request by VadimGrznk**
Add using recently added global `--output` argument to` --sign-only` transactions.

Fixes #8281

### CODE REVIEW

File: cli/src/cli_output.rs [hunk 1]
```diff
@@ -839,3 +840,53 @@ impl From<&Lockout> for CliLockout {
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct CliSignOnlyCommand {
```
Comment (CriesofCarrots) [inline cli/src/cli_output.rs:845]: Let's rename this slightly to describe the contents, not the caller. How about `CliSignOnlyData`?

File: cli/src/cli.rs [hunk 2]
```diff
@@ -1062,15 +1062,16 @@ pub fn return_signers(tx: &Transaction) -> ProcessResult {
             }
         });
 
-    println_signers(&tx.message.recent_blockhash, &signers, &absent, &bad_sig);
+    let cli_command = CliSignOnlyData {
+        blockhash: tx.message.recent_blockhash,
+        signers: signers,
+        absent: absent,
+        bad_sig: bad_sig,
```
Comment (CriesofCarrots) [inline cli/src/cli.rs:1069]:
```suggestion
        signers,
        absent,
        bad_sig,
```

**Comment by CriesofCarrots** [comment]
@VadimGrznk , can you please rebase on master to pull in the fix for that CI failure?
CI should now run automatically on this PR, so you can check to see if there are any other issues.

**Comment by CriesofCarrots** [comment]
Incidentally, it looks like `test_return_signers()` fails because `return_signers` no longer returns the signer data.
Perhaps add an internal method for unit testing:
```
fn _return_signers(tx: &Transaction, config: &CliConfig) -> CliSignOnlyData {
  ...
}
```

And `return_signers` becomes simply:
```
pub fn return_signers(tx: &Transaction, config: &CliConfig) -> ProcessResult {
    let signers = _return_signers(tx, config);
    config.output_format.formatted_print(&signers);
    Ok("".to_string())
```

**Comment by VadimGrznk** [comment]
@CriesofCarrots Hello
I decided to solve it in a little different way because couple of other tests also expected json string as a result of command processing. So now `return_signers` always return json string in `Result` enum as earlier but that string is not printing it's just for tests. I found such solution the most appropriate in such case. What do you think?

File: cli/src/offline/mod.rs [hunk 3]
```diff
@@ -79,32 +79,56 @@ pub fn parse_sign_only_reply_string(reply: &str) -> SignOnly {
...
-    let bad_signers = signer_strings
-        .iter()
-        .map(|val| {
-            let s = val.as_str().unwrap();
-            Pubkey::from_str(s).unwrap()
-        })
-        .collect();
+    let mut present_signers: Vec<(Pubkey, Signature)> = Vec::new();
+    let signer_strings = object.get("signers");
+    match signer_strings {
```
Comment (CriesofCarrots) [inline cli/src/offline/mod.rs:84]: Clippy doesn't like these matches. There are a couple, more Rust-y, approaches to take here. I think the simplest is probably `.map(...).unwrap_or_default()`, eg:

```
let present_signers = object
        .get("signers")
        .map(|sig_strings| {
            sig_strings
                .as_array()
                .unwrap()
                .iter()
                .map(|signer_string| {
                    let mut signer = signer_string.as_str().unwrap().split('=');
                    let key = Pubkey::from_str(signer.next().unwrap()).unwrap();
                    let sig = Signature::from_str(signer.next().unwrap()).unwrap();
                    (key, sig)
                })
                .collect()
        })
        .unwrap_or_default();
```

**Comment by CriesofCarrots** [comment]
Thank you, @VadimGrznk ! I rebased this for you

**Pull request merged**